lib/widgets/action_sheet.dart

```diff
@@ -36,20 +36,46 @@ void showMessageActionSheet({required BuildContext context, required Message mes
       && reactionWithVotes.userIds.contains(store.selfUserId))
     ?? false;
 
-  showDraggableScrollableModalBottomSheet<void>(
+  final designVariables = DesignVariables.of(context);
+  showModalBottomSheet<void>(
     context: context,
+    clipBehavior: Clip.antiAlias,
+    backgroundColor: designVariables.actionSheetBackground,
+    shape: const RoundedRectangleBorder(borderRadius: BorderRadius.vertical(top: Radius.circular(20.0))),
+    useSafeArea: true,
+    isScrollControlled: true,
     builder: (BuildContext _) {
-      return Column(children: [
-        if (!hasThumbsUpReactionVote) AddThumbsUpButton(message: message, messageListContext: context),
-        StarButton(message: message, messageListContext: context),
-        if (isComposeBoxOffered) QuoteAndReplyButton(
-          message: message,
-          messageListContext: context,
+      return Padding(
+        padding: const EdgeInsets.all(16.0),
+        child: Column(
+          crossAxisAlignment: CrossAxisAlignment.stretch,
+          mainAxisSize: MainAxisSize.min,
+          children: [
+            // TODO(#217): show message text
+            Flexible(
+              child: SingleChildScrollView(
+                child: ClipRRect(
+                  borderRadius: BorderRadius.circular(7),
+                  child: Column(children: [
+                    if (!hasThumbsUpReactionVote) AddThumbsUpButton(message: message, messageListContext: context),
+                    StarButton(message: message, messageListContext: context),
+                    if (isComposeBoxOffered) QuoteAndReplyButton(
+                      message: message,
+                      messageListContext: context,
+                    ),
+                    CopyMessageTextButton(message: message, messageListContext: context),
+                    CopyMessageLinkButton(message: message, messageListContext: context),
+                    ShareButton(message: message, messageListContext: context),
+                  // TODO: The following line could be replaced by the [spacing]
+                  // property when https://github.com/flutter/flutter/issues/55378 is fixed.
+                  ].expand((item) => [const SizedBox(height: 1), item]).skip(1).toList()),
+                ),
+              ),
+            ),
+            const MessageActionSheetCancelButton(),
```

Comment:
There needs to be 8px padding between Menu buttons and Cancel button as per design:

<img width="849" alt="Screenshot 2024-08-06 at 21 52 34" src="https://github.com/user-attachments/assets/feb06eab-3d1a-473e-ae3f-db285ea96a01">